File: Q15.cpp
#include <iostream>
#include <stdio.h>

using namespace std;

int main()
{
    float numero[20];
    int vezes = 1, quantidade = 0;
    for (int i = 0; i < sizeof numero / sizeof numero[0]; i ++, vezes ++)
    {
        printf("Digite o %d° número: ", vezes);
        cin >> numero[i];
        if (numero[i] >= 0 & numero[i] <= 100)
        {
            quantidade ++;
        }
    }
    switch (quantidade)
    {
        case 0:    
            cout << "Dos 20 números que você digitou, nenhum deles está entre 0 e 100." << endl;
            break;
        case 20:
            cout << "Dos 20 números que você digitou, todos eles estão entre 0 e 100." << endl;
            break;
        default:
            printf("Dos 20 números que você digitou, %d deles estão entre 0 e 100.\n", quantidade);
            break;
    }
    return 0;
}

File: Q15.py
número = list(range(20))
vezes = 1
quantidade = 0
for i in range(0, len(número)):
    número[i] = float(input(f"Digite o {vezes}° número: "))
    if número[i] >= 0 and número[i] <= 100:
        quantidade += 1
    vezes += 1
match quantidade:
    case 0:
        print("Dos 20 números que você digitou, nenhum deles está entre 0 e 100.")
    case 20:
        print("Dos 20 números que você digitou, todos eles estão entre 0 e 100.")
    case _:
        print(f"Dos 20 números que você digitou, {quantidade} deles estão entre 0 e 100.")